### PR TITLE
Add the `nointerface` attribute

### PR DESCRIPTION
**Pull request by rachitnigam**
Useful when generating low-level or non-invokable Calyx components

### CODE REVIEW

File: tests/parsing/nointerface.expect [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import "primitives/core.futil";
```
Comment (sampsyo) [inline tests/parsing/nointerface.expect:1]: It looks like maybe the input for this test is missing?

File: docs/lang/attributes.md [hunk 2]
```diff
@@ -51,6 +51,15 @@ The entrypoint for the Calyx program. If no component has this attribute, then
...
 compiler errors out.
 
+### `go`, `done`, `clk` and `reset`
+These four ports are part of the interface to Calyx components.
+`go` and `done` provide the mechanism for how an "outer" component invokes an "inner" cell that it contains.
+`clk` and `reset` thread through the global clock and resetting signal in a design.
+
+### `nointerface`
+By default, interface ports are automatically added to a component if they are missing.
+Adding this attribute disables this behavior.
```
Comment (sampsyo) [inline docs/lang/attributes.md:61]: Could be worth pointing out _who_ adds interface ports—namely, this happens during parsing (not, like, a special lowering pass).

**Review by sampsyo — COMMENTED** [review]
Just noting a missing test (and a possible docs suggestion).

File: docs/lang/attributes.md (hunk 2 above)
Comment (rachitnigam) [reply docs/lang/attributes.md:61]: These are roundtripping tests so the input file is the same as the output file

File: tests/parsing/nointerface.expect (hunk 1 above)
Comment (rachitnigam) [reply tests/parsing/nointerface.expect:1]: These are roundtripping tests so the input file is the same as the output file

Comment (sampsyo) [reply tests/parsing/nointerface.expect:1]: Oh awesome; thanks for clarifying! Cool setup.

**Pull request merged**